preferencesfx/src/main/java/com/dlsc/preferencesfx/util/StorageHandlerImpl.java

```diff
@@ -25,7 +27,21 @@ protected String serialize(Object object) {
   }
 
   @Override
-  protected <T> T deserialize(String serialized, Class<? extends T> type) {
+  protected <T> T deserialize(String serialized, Class<T> type) {
     return gson.fromJson(serialized, type);
   }
+
+  @Override
+  protected <T> List<T> deserializeList(String serialized, Class<T> type) {
+    final TypeToken typeToken = new TypeToken<List<JsonElement>>() {};
+    final List<JsonElement> list = gson.fromJson(serialized, typeToken.getType());
+
+    if (list == null) {
+      return null;
```

Comment:
Good catch.
But should we really convert a value which is stored as null to an empty list??

Reply:
I think it wouldn't make sense later in the process needing to differentiate whether a setting value loaded was null or empty. Here it's more about what we would expect: do we expect to always get a list when calling this method? I think it would make it easier later, since we have to pay less attention at handling `null`, which is why I think this is quite a sensible suggestion.
Or is there anything you could think of that could pose an issue with that?

Reply:
I mainly implemented it this way so it behaves like the previous implementation

Reply:
Ah, I see! I think it should be okay, but I would of course test it first!

Reply:
Hard to say how to test as the value (null or emptyList) will get passed to a ListProperty coming from the client code. So I do not know what behavior the client expects as I do not have all the clients code...

Reply:
I will have a look at it and get back to you!

Reply:
`deserializeList` is only being called in `PreferencesBasedStorageHandler`, wrapped in a `FXCollections.observableArrayList(...)`, which in itself should not be `null`. So I think we should return an empty collection here.

Reply:
ack. returning null will trigger a NPE in `loadObservableList` method.

Fixed it.